homeworks/Assignment1_Final.R

```diff
@@ -0,0 +1,209 @@
+
+###############################################################################################
+# Loading Libraries
+# install.packages as necessary
+###############################################################################################
+library(lintr)
+library(styler)
+library(tidyverse)
+library(stats)
+library(dbplyr)
+library(data.table)
+###############################################################################################
+#Problem 1
+###############################################################################################
+# (1) 
+a <- seq(0, 100, by = 1)
+a
+typeof(a)
+#This is a double atomic vector
+a
+b <- seq_len(100) #c(1:100) #These are integer atomic vectors
+b
+typeof(a)
+typeof(b)
+is.atomic(a)
+
+# (2)
+c <- rep(10, times = 100)
+c
+typeof(c)
+
+# (3)
+d <- rep(c(1, 2, 3, 4, 5), times = 10)
+d
+typeof(d)
+
+# (4)
+d <- rep(c(1, 2, 3, 4, 5), each = 10)
+d
+typeof(d)
+
+# (5)
+e <- seq(0, 1, by = 0.1)
+e
+typeof(e)
+
+###############################################################################################
+#Problem 2
+###############################################################################################
+
+help('rivers')
+data('rivers')
+View(data)
+# (1) 
+## This type of data is an Atomic vector, as it only stores one consistent element, numbers, In this case legnths in miles(One dimensional). Since the data only presents numbers, this particular data can be characterized as either 'double' or 'integer' atomic vectors.
+
+# (2)
+##Calculations
+data(rivers)
+me <- log(mean(rivers)) #Mean
+sd <- log(sd(rivers)) #Standard Deviation
+ma <- log(max(rivers)) #Maximum Value
+mi <- log(min(rivers)) #Minimum Value
+le <- log(length(rivers)) #Length
+va <- log(var(rivers)) #Variance
+##Creating Vector
+vector <- c(me, sd, ma, mi, le, va)
+vector
+##Final Vector, Naming
+names(vector) <- c("Mean", "Standard Deviation", "Maximum", "Minimum", "Length", "Variance")
+vector
+##Checking for desired parameters
+is.atomic(vector) #TRUE
+str(vector)
+
+# (3)
+##Adjusting Data
+x <- sort(rivers)
+x
+x2 <- x[!x %in% c(135:237, 1270:3710)]
+x2
+##New Calculations
+me <- log(mean(x2)) #Mean
+sd <- log(sd(x2)) #Standard Deviation
+ma <- log(max(x2)) #Maximum Value
+mi <- log(min(x2)) #Minimum Value
+le <- log(length(x2)) #Length
+va <- log(var(x2)) #Variance
+##Creating Vector
+vector_new <- c(me, sd, ma, mi, le, va)
+vector_new
+##Final Vector, Naming
+names(vector_new) <- c("Mean", "Standard Deviation", "Maximum", "Minimum", "Length", "Variance")
+vector_new
+##Checking for desired parameters
+is.atomic(vector_new) #TRUE
+str(vector_new)
+
+##Checking both vectors
+vector
+vector_new
+
+###############################################################################################
+#Problem 3
+###############################################################################################
+# (1)
+x <- c(5, 6, 7, 8)
+y <- c("a", "b", "c", "d")
+u <- list(x, y)
+str(u)
+ 
+# (2)
+u[[2]] <- c(1, 2, 3, 4)
```

Comment:
same here, try to not to use the position, but the name will make your code more robust :)